bin/package/config/darwin/update-resolv-conf

```diff
@@ -39,44 +39,65 @@ update_all_dns()
 {
     for adapter in "${adapters[@]}"
     do
-    echo updating dns for $adapter
-    # set dns server to the vpn dns server
-    if [[ "${SRCHS[@]}" ]]; then
-        $NETWORKSETUP -setsearchdomains "$adapter" "${SRCHS[@]}"
-    fi
-    if [[ "${NMSRVRS[@]}" ]]; then
-        $NETWORKSETUP -setdnsservers "$adapter" "${NMSRVRS[@]}"
-    fi
+        echo updating dns for $adapter
+        # set dns server to the vpn dns server
+        if [[ "${SRCHS[@]}" ]]; then
+            $NETWORKSETUP -getsearchdomains "$adapter" | grep -v "There aren't any" > "/tmp/myst.$adapter.searchdomains"
+            $NETWORKSETUP -setsearchdomains "$adapter" "${SRCHS[@]}"
+        fi
+        if [[ "${NMSRVRS[@]}" ]]; then
+            $NETWORKSETUP -getdnsservers "$adapter" | grep -v "There aren't any" > "/tmp/myst.$adapter.dnsservers"
+            $NETWORKSETUP -setdnsservers "$adapter" "${NMSRVRS[@]}"
+        fi
     done
 }
 
 clear_all_dns()
 {
     for adapter in "${adapters[@]}"
     do
-    echo updating dns for $adapter
-    $NETWORKSETUP -setdnsservers "$adapter" empty
-    $NETWORKSETUP -setsearchdomains "$adapter" empty
+        echo updating dns for $adapter
+
+        dnsservers=""
+        searchdomains=""
```

Comment:
do we need this extra initialization here? We are doing it right below.

Reply:
It's a loop, it preserves value from the previous iteration. Resetting it explicitly.